lib/fastimage.rb

```diff
@@ -396,56 +437,49 @@ def fetch_using_read(readable)
       end
     end
 
-    parse_packets FiberStream.new(read_fiber)
+    parse_packets FiberStream.new(read_fiber), property
   end
 
-  def fetch_using_file_open
-    @content_length = File.size?(@uri)
+  def fetch_using_file_open(property)
+    return @content_length = File.size?(@uri) if property == :content_length
```

Comment:
It would seem a bit neater to me to always set @content_length here - and after that do the early return if content length was being asked for

Reply:
Ah ok!